### ISSUE TITLE
TopTenData not generated

### ISSUE THREAD

**Issue by YellowZaki** [BentoBoxWorld/Level#93]
**Description**
Since some versions ago, I noticed that the TopTen folder wasn't updating.
I deleted LevelsData and TopTenData. Both folders were generated again, but TopTenData didn't contain anything so top wasn't persistent between server restarts.

**Server Information:**
Paper 1.13.2

[Please complete the following information:]
 - Database being used: YAML
 - BentoBox version: [e.g. 1.7.2.21]

**Comment by tastybento** [comment]
Hmm, I can't replicate this on 1.8.0 BentoBox and Level. I even tried YAML and it was okay. (Are you still on YAML? I recommend upgrading to JSON using YAML2JSON database).

What files do you see in the database/TopTenData folder? There should be one per world, like bskyblock_world.yml for example.

Can you share your BSkyBlock config?

**Comment by YellowZaki** [comment]
There is nothing on TopTenData folder. I've tried deleting LevelsData and TopTenData several times and they are regenerated, but TopTenData empty.
LevelsData gets filled fine.

BSkyBlock config: https://pastebin.com/bWkivd6S

PD: I'm going to migrate YAML2JSON

**Comment by YellowZaki** [comment]
I've migrated to json and top ten file still not generated.

**Comment by tastybento** [comment]
Okay weird. I’ll try with the versions you are using and see. 
I assume you are just doing /is level to make an entry and the top ten are shown in the game? If not, how are you making it?

**Comment by YellowZaki** [comment]
Yes, /is topten is shown ingame when users do /is level. But on server restart, is topten is empty again.

**Comment by tastybento** [comment]
I tried with your configs and the server/addon versions and everything checks out.

I had a look at your level config though and you require 704 points per level and the only items you give points to are acacia doors and buttons. Isn't that weird? It's likely that no one has any levels at all if that is the case so the top ten will be zero. Could that be the problem?

The only other thing I can think of is that there must be a console error that is occurring when you do an island level or shutdown. Have a look and see if you can find anything. Apart from that, sorry but I can't find a problem.

**Comment by YellowZaki** [comment]
I didn't post the full list of blocks, levels are calculated perfectly.
I'll try on a test server without other addons and I'll let you know. Thank you.

**Comment by YellowZaki** [comment]
Ok, I found why, it was my fault. It is because a custom addon I left:

```
    @Override
    public void onDisable() {
        throw new UnsupportedOperationException("Not supported yet."); 
    }
```
So when BentoBox shut down, the exception was fired and it prevented TopTen from saving.
I was lucky it was only affecting Level addon.

Maybe this info could be useful to let you know that if a exception occurs on addon disable, it prevents saving. Maybe something can be done in order to prevent that issue.

**Comment by tastybento** [comment]
Okay, thanks. Yes, I'll look into this.

**Comment by Poslovitch** [comment]
Should I open a ticket on BentoBox's bug tracker so that we don't forget to do that?
